static/src/javascripts/projects/commercial/modules/creatives/page-skin.js

```diff
@@ -61,76 +60,87 @@ const pageSkin = (): void => {
-                }
-
-                if (window.pageYOffset === 0) {
-                    moveBackgroundVerticalPosition(topPosition);
-                }
-
-                const headerBoundaries = header.getBoundingClientRect();
-                const topBannerAdBoundaries = topBannerAd.getBoundingClientRect();
-                const headerPosition = headerBoundaries.top;
-                const topBannerBottom = topBannerAdBoundaries.bottom;
-                const fabricScrollStartPosition =
-                    topBannerAdBoundaries.height +
-                    adLabelHeight -
-                    headerBoundaries.height;
-
-                if (
-                    headerPosition <= fabricScrollStartPosition &&
-                    topBannerBottom > 0
-                ) {
-                    moveBackgroundVerticalPosition(topBannerBottom);
-                } else if (topBannerBottom <= 0) {
-                    moveBackgroundVerticalPosition(0);
-                }
-            }
-            // This is to reposition the Page Skin to start where the navigation header ends.
-            if (!hasTruskin && hasPageSkin && isInAUEdition) {
-                initTopPositionOnce(false);
-                if (window.pageYOffset === 0) {
-                    moveBackgroundVerticalPosition(topPosition);
-                } else if (window.pageXOffset <= topPosition) {
-                    moveBackgroundVerticalPosition(
-                        topPosition - window.pageYOffset
-                    );
-                }
-                if (window.pageYOffset > topPosition) {
-                    moveBackgroundVerticalPosition(0);
-                }
-            }
+            moveBackgroundVerticalPosition(topBannerBottom);
+        } else if (topBannerBottom <= 0) {
+            moveBackgroundVerticalPosition(0);
+        }
+    };
+
+    const repositionPageSkin = (): void => {
+        initTopPositionOnce();
+        if (window.pageYOffset === 0) {
+            moveBackgroundVerticalPosition(topPosition);
+        } else if (window.pageXOffset <= topPosition) {
+            moveBackgroundVerticalPosition(topPosition - window.pageYOffset);
+        }
+        if (window.pageYOffset > topPosition) {
+            moveBackgroundVerticalPosition(0);
+        }
+    };
+
+    const repositionSkin = (): void => {
+        const header = document.querySelector('.new-header');
+        const footer = document.querySelector('.l-footer');
+        const topBannerAd = document.querySelector('.ad-slot--top-banner-ad');
+
+        if (hasTruskin && header && topBannerAd && footer) {
+            repositionTruskin(header, footer, topBannerAd);
+        }
+        // This is to reposition the Page Skin to start where the navigation header ends.
+        if (!hasTruskin && hasPageSkin && isInAUEdition) {
+            repositionPageSkin();
         }
     };
 
     togglePageSkin();
 
+    const removeTopAdBorder = (): void => {
```

Comment:
Would it be simpler to put this logic into the `repositionTruskin` function, so that the code is all in one place?

Reply:
We originally had it the way you suggest but there was a race condition that would cause the border to be removed after the repositioning calculations and that would leave behind a 1 pixel gap in the image.

There might be a specific place within the `repositionTruskin` where it still works, but placing it outside guaranteed that the border would be removed before calculations.